src/main/java/TaskList.java

```diff
@@ -0,0 +1,201 @@
+        printDottedLines();
+    }
+
+    private String checkAndMarkTaskAsDone(int index) {
+        String phrase;
+        if((index >= tasksCounter)||(index < 0)){
+            phrase = getInvalidTaskPhrase();
+        }
+        else if (this.isDone[index]) {
+            phrase = getTaskAlreadyCompletedPhrase();
+
+        }
+        else{
+            (this.isDone)[index] = true;
+            areAllTasksNotDone = false;
+            phrase = getTaskMarkedAsDoneMessage(index);
+        }
+        return phrase;
+    }
+
+    private String getTaskMarkedAsDoneMessage(int index) {
+        String phrase;
+        if (getAreAllTasksDone()) {
+            phrase ="Nice! I've marked this task as done:"+ System.lineSeparator()
+                    +(index +1)+"["+getStatusIcon(index)+"]"+". "+getTaskDescription(index)+" "+System.lineSeparator()
+                    + "Good job Crewmate! You completed all your tasks! (─‿─) in this list"+System.lineSeparator()
+                    + "Don't forget to check the other lists as well!!!!" ;
+        }
+        else {
+            phrase =" Nice! I've marked this task as done:"+ System.lineSeparator()
+                    +(index +1)+"["+getStatusIcon(index)+"]"+". "+getTaskDescription(index)+" "+System.lineSeparator()
+                    + "You still have " + getNumberTasksLeft()
+                    + " task(s) left in this list Crewmate! Hurry up!! ＼(｀0´)／";
+
+        }
+        return phrase;
+    }
+
+    public String getTaskAlreadyCompletedPhrase() {
+        String phrase;
+        if(getAreAllTasksDone()){
+            phrase = "This job was already completed!!!!"+System.lineSeparator()
+                    + "Good job Crewmate! You completed all your tasks in this list! (─‿─)"+System.lineSeparator()
+                    + "Don't forget to check the other lists as well!!!!";
+
+
+        }
+        else{
+            phrase ="What are you doing??? This job was already completed!! (;¬_¬)"+System.lineSeparator()
+                    + "You still have " + getNumberTasksLeft()
+                    + " tasks left in this list Crewmate! Hurry up!! ＼(｀0´)／";
+        }
+        return phrase;
+    }
+
+    private String getInvalidTaskPhrase() {
+        String phrase;
+        if(getAreAllTasksNotDone()) {
+            phrase ="There's no such task?! Are you really a Crewmate??  (╬⓪益⓪)";
+        }
+        else{
+            phrase = "There's no such task?! Focus Crewmate!!  (╬⓪益⓪)";
+        }
+        return phrase;
+    }
+
+    public boolean getIsTaskDone(int index){
+        return (this.isDone)[index];
+    }
+
+    public String getTaskDescription(int index){
+        return (this.description)[index];
+    }
+
+    public int getNumberTasksLeft(){
+        if(getAreAllTasksDone()){
+            return 0;
+        }
+        else if(getAreAllTasksNotDone()){
+            return getTasksCounter();
+        }
+        else{
+            return calculateNumberOfTasksLeft();
```

Comment:
If I remember correctly from the course learning materials, if-else statements should have the form:

if (condition) {
    statements;
} else {
    statements;
}

Reply:
You are right! Thanks